a10_octavia/network/drivers/neutron/a10_octavia_neutron.py

```diff
@@ -368,10 +368,10 @@ def allocate_vrid_fip(self, vrid, network_id, amphorae, fixed_ip=None):
     def _remove_allowed_address_pair_from_port(self, port_id, ip_address):
         port = self.neutron_client.show_port(port_id)
         aap_ips = port['port']['allowed_address_pairs']
-        aap_ips.remove({'ip_address': ip_address})
+        updated_aap_ips = [aap_ip for aap_ip in aap_ips if aap_ip['ip_address'] != ip_address]
```

Comment:
If ip_address already does not exist then there is no need to remove it i.e, making call for update_port.